# Customer Churn Prediction & Retention Analytics Platform

## SQL Database Preparation

### Objective

Load the cleaned customer churn dataset into a SQLite database.

The database will be used for SQL-based business analysis in the next phase of the project.

### Output

- SQLite Database (`churn.db`)
- Table (`customer_churn_data`)

In [1]:

import pandas as pd
import sqlite3

In [2]:
sql_df = pd.read_csv(
    "../data/cleaned/Telco_customer_churn_clean.csv"
)

sql_df.head()

,CustomerID,City,Zip Code,Lat Long,Latitude,Longitude,Gender,Senior Citizen,Partner,Dependents,...,Contract,Paperless Billing,Payment Method,Monthly Charges,Total Charges,Churn Label,Churn Value,Churn Score,CLTV,Churn Reason
0,3668-QPYBK,Los Angeles,90003,"33.964131, -118.272783",33.964131,-118.272783,Male,No,No,No,...,Month-to-month,Yes,Mailed check,53.85,108.15,Yes,1,86,3239,Competitor made better offer
1,9237-HQITU,Los Angeles,90005,"34.059281, -118.30742",34.059281,-118.307420,Female,No,No,Yes,...,Month-to-month,Yes,Electronic check,70.70,151.65,Yes,1,67,2701,Moved
2,9305-CDSKC,Los Angeles,90006,"34.048013, -118.293953",34.048013,-118.293953,Female,No,No,Yes,...,Month-to-month,Yes,Electronic check,99.65,820.5,Yes,1,86,5372,Moved
3,7892-POOKP,Los Angeles,90010,"34.062125, -118.315709",34.062125,-118.315709,Female,No,Yes,Yes,...,Month-to-month,Yes,Electronic check,104.80,3046.05,Yes,1,84,5003,Moved
4,0280-XJGEX,Los Angeles,90015,"34.039224, -118.266293",34.039224,-118.266293,Male,No,No,Yes,...,Month-to-month,Yes,Bank transfer (automatic),103.70,5036.3,Yes,1,89,5340,Competitor had better devices


In [3]:
print(f"Rows: {sql_df.shape[0]}")
print(f"Columns: {sql_df.shape[1]}")

Rows: 7043
Columns: 30


In [4]:
sql_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 30 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   CustomerID         7043 non-null   str    
 1   City               7043 non-null   str    
 2   Zip Code           7043 non-null   int64  
 3   Lat Long           7043 non-null   str    
 4   Latitude           7043 non-null   float64
 5   Longitude          7043 non-null   float64
 6   Gender             7043 non-null   str    
 7   Senior Citizen     7043 non-null   str    
 8   Partner            7043 non-null   str    
 9   Dependents         7043 non-null   str    
 10  Tenure Months      7043 non-null   int64  
 11  Phone Service      7043 non-null   str    
 12  Multiple Lines     7043 non-null   str    
 13  Internet Service   7043 non-null   str    
 14  Online Security    7043 non-null   str    
 15  Online Backup      7043 non-null   str    
 16  Device Protection  7043 non-null   

In [5]:
conn = sqlite3.connect("../database/churn.db")

print("SQLite Database Connected Successfully!")

SQLite Database Connected Successfully!


In [6]:
query = """
SELECT *
FROM customer_churn
LIMIT 5;
"""

pd.read_sql(query, conn)

,CustomerID,City,Zip Code,Lat Long,Latitude,Longitude,Gender,Senior Citizen,Partner,Dependents,...,Contract,Paperless Billing,Payment Method,Monthly Charges,Total Charges,Churn Label,Churn Value,Churn Score,CLTV,Churn Reason
0,3668-QPYBK,Los Angeles,90003,"33.964131, -118.272783",33.964131,-118.272783,Male,No,No,No,...,Month-to-month,Yes,Mailed check,53.85,108.15,Yes,1,86,3239,Competitor made better offer
1,9237-HQITU,Los Angeles,90005,"34.059281, -118.30742",34.059281,-118.307420,Female,No,No,Yes,...,Month-to-month,Yes,Electronic check,70.70,151.65,Yes,1,67,2701,Moved
2,9305-CDSKC,Los Angeles,90006,"34.048013, -118.293953",34.048013,-118.293953,Female,No,No,Yes,...,Month-to-month,Yes,Electronic check,99.65,820.5,Yes,1,86,5372,Moved
3,7892-POOKP,Los Angeles,90010,"34.062125, -118.315709",34.062125,-118.315709,Female,No,Yes,Yes,...,Month-to-month,Yes,Electronic check,104.80,3046.05,Yes,1,84,5003,Moved
4,0280-XJGEX,Los Angeles,90015,"34.039224, -118.266293",34.039224,-118.266293,Male,No,No,Yes,...,Month-to-month,Yes,Bank transfer (automatic),103.70,5036.3,Yes,1,89,5340,Competitor had better devices


In [9]:
sql_df.to_sql(
    name="customer_churn_data",
    con=conn,
    if_exists="replace",
    index=False
)

print("Table Created Successfully!")

Table Created Successfully!


In [10]:
query = """
SELECT *
FROM customer_churn_data
LIMIT 5;
"""

pd.read_sql(query, conn)

,CustomerID,City,Zip Code,Lat Long,Latitude,Longitude,Gender,Senior Citizen,Partner,Dependents,...,Contract,Paperless Billing,Payment Method,Monthly Charges,Total Charges,Churn Label,Churn Value,Churn Score,CLTV,Churn Reason
0,3668-QPYBK,Los Angeles,90003,"33.964131, -118.272783",33.964131,-118.272783,Male,No,No,No,...,Month-to-month,Yes,Mailed check,53.85,108.15,Yes,1,86,3239,Competitor made better offer
1,9237-HQITU,Los Angeles,90005,"34.059281, -118.30742",34.059281,-118.307420,Female,No,No,Yes,...,Month-to-month,Yes,Electronic check,70.70,151.65,Yes,1,67,2701,Moved
2,9305-CDSKC,Los Angeles,90006,"34.048013, -118.293953",34.048013,-118.293953,Female,No,No,Yes,...,Month-to-month,Yes,Electronic check,99.65,820.5,Yes,1,86,5372,Moved
3,7892-POOKP,Los Angeles,90010,"34.062125, -118.315709",34.062125,-118.315709,Female,No,Yes,Yes,...,Month-to-month,Yes,Electronic check,104.80,3046.05,Yes,1,84,5003,Moved
4,0280-XJGEX,Los Angeles,90015,"34.039224, -118.266293",34.039224,-118.266293,Male,No,No,Yes,...,Month-to-month,Yes,Bank transfer (automatic),103.70,5036.3,Yes,1,89,5340,Competitor had better devices


In [11]:
query = """
SELECT COUNT(*) AS Total_Customers
FROM customer_churn_data;
"""

pd.read_sql(query, conn)

,Total_Customers
0,7043


In [12]:
query = """
SELECT name
FROM sqlite_master
WHERE type='table';
"""

pd.read_sql(query, conn)

,name
0,customer_churn_data


In [13]:
conn.close()

# SQL Preparation Summary

Database Successfully Created

Database Name:
churn.db

Table Name:
customer_churn_data

Total Records:
7043

Total Features:
30

The SQLite database is now ready for SQL business analysis.